plugins/modules/files/iso_customize.py

```diff
@@ -0,0 +1,432 @@
+     type: list
+     elements: dict
+     required: no
+     suboptions:
+       src_file:
+         description:
+         - The absolute path with file name in local machine.
+         type: path
+       dest_file:
+         description:
+         - The absolute path with file name on ISO file.
+         type: path
+   delete_files:
+     description:
+     - The absolute path with file name on ISO file.
+     type: list
+     elements: path
+     required: no
+   modify_files:
+     description:
+     - replace the line/lines with specific string in file of ISO
+     type: list
+     elements: dict
+     required: no
+     suboptions:
+       file:
+         description:
+         - The absolute path with file name on ISO file.
+         type: path
+       regexp:
+         description:
+         - the match string in line of file, it supports regular expression
+         type: str
+       replace:
+         description:
+         - the target string
+         type: str
+'''
+
+EXAMPLES = r'''
+- name: "Customize ISO file"
+  community.general.iso_customize:
+    src_iso: "/Users/zouy/Documents/ubuntu-22.04-desktop-amd64.iso"
+    dest_iso: "/Users/zouy/Documents/ubuntu-22.04-desktop-amd64-customized.iso"
+    delete_files:
+      - "/preseed/ubuntu.seed"
+    add_files:
+      - src_file: "/Users/zouy/Documents/ks.cfg"
+        dest_file: "/ks.cfg"
+      - src_file: "/Users/zouy/Documents/ubuntu.seed"
+        dest_file: "/preseed/ubuntu.seed"
+    modify_files:
+      - file: "/boot/grub/grub.cfg"
+        regexp: "timeout=30"
+        replace: "default=0\nset timeout=5"
+      - file: "/boot/grub/grub.cfg"
+        regexp: "maybe-ubiquity quiet splash ---"
+        replace: 'boot=casper debug-ubiquity automatic-ubiquity quiet splash noprompt ---'
+  register: customize_iso_result
+'''
+
+RETURN = r'''
+src_iso:
+    description: path of source ISO file.
+    returned: on success
+    type: str
+    sample: "/path/to/file.iso"
+dest_iso:
+    description: path of the customized iso file.
+    returned: on success
+    type: str
+    sample: "/path/to/customized.iso"
+
+'''
+
+import os
+import traceback
+import re
+import shutil
+import time
+import copy
+
+PYCDLIB_IMP_ERR = None
+try:
+    import pycdlib
+    HAS_PYCDLIB = True
+except ImportError:
+    PYCDLIB_IMP_ERR = traceback.format_exc()
+    HAS_PYCDLIB = False
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+from ansible.module_utils.common.text.converters import to_native
+
+MODULE_ISO_CUSTOMIZE = None
+
+
+# Create local temporary direction to store files getting from ISO
+def get_local_tmp_dir():
+    dir_name = time.strftime('%Y-%m-%d-%H-%M-%S', time.localtime(time.time()))
+    tmp_dir = os.path.join('/tmp', dir_name)
+
+    try:
+        os.makedirs(tmp_dir)
+    except OSError as err:
+        msg = "Failed to create folder %s with error %s" % (tmp_dir, to_native(err))
+        MODULE_ISO_CUSTOMIZE.fail_json(msg=msg)
+
```

Comment:
You should use the functions in `tempfile` instead:
https://docs.python.org/3/library/tempfile.html

Reply:
Thanks. It's changed.